Quasardb.Tests/Query/Query.cs

```diff
@@ -0,0 +1,338 @@
+            for (int i = 0; i < count; ++i)
+            {
+                r[i] = new byte[32];
+                random.NextBytes(r[i]);
+                column.Insert(time, r[i]);
+                time = time.AddSeconds(1);
+            }
+            return r;
+        }
+
+        public double[] InsertDoublePoints(string alias, DateTime time, int count)
+        {
+            Random random = new Random();
+            var r = new double[count];
+
+            var ts = _cluster.TimeSeries(alias);
+            var column = ts.DoubleColumns["the_double"];
+            for (int i = 0; i < count; ++i)
+            {
+                column.Insert(time, r[i] = random.NextDouble());
+                time = time.AddSeconds(1);
+            }
+            return r;
+        }
+
+        public long[] InsertInt64Points(string alias, DateTime time, int count)
+        {
+            Random random = new Random();
+            var r = new long[count];
+
+            var ts = _cluster.TimeSeries(alias);
+            var column = ts.Int64Columns["the_int64"];
+            for (int i = 0; i < count; ++i)
+            {
+                column.Insert(time, r[i] = random.Next());
+                time = time.AddSeconds(1);
+            }
+            return r;
+        }
+
+        public DateTime[] InsertTimestampPoints(string alias, DateTime time, int count)
+        {
+            Random random = new Random();
+            var r = new DateTime[count];
+
+            var ts = _cluster.TimeSeries(alias);
+            var column = ts.TimestampColumns["the_ts"];
+            for (int i = 0; i < count; ++i)
+            {
+                column.Insert(time, r[i] = DateTime.Today.AddSeconds(random.NextDouble()));
+                time = time.AddSeconds(1);
+            }
+            return r;
+        }
+
+        private void CheckColumns(QdbColumnNameCollection columns)
+        {
+            Assert.AreEqual(6, columns.Count);
+            Assert.AreEqual("$timestamp", columns[0]);
+            Assert.AreEqual("$table", columns[1]);
+            Assert.AreEqual("the_blob", columns[2]);
+            Assert.AreEqual("the_double", columns[3]);
+            Assert.AreEqual("the_int64", columns[4]);
+            Assert.AreEqual("the_ts", columns[5]);
+        }
+
+        #region Query failure tests
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void ThrowsForNullQuery()
+        {
+            _cluster.Query(null);
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(QdbQueryException))]
+        public void ThrowsForEmptyQuery()
+        {
+            _cluster.Query("");
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(QdbQueryException))]
+        public void ThrowsForInvalidQuery()
+        {
+            _cluster.Query("select * from");
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(QdbQueryException))]
```

Comment:
Should that be a query exception? Don't we already have alias-not-found exception or sth?

Reply:
I initially went with `QdbAliasNotFoundException` (and all the existing others), but I could not find a way to extract the alias name. Since it is present in the `error_message` returned from query, I augmented all exception machinery to optionally take an explicit message to override the error specific one. The result ended up being confusing, I would try to use properties like `e.Alias` in tests but those of course continued to be `null`.

Reply:
Ok.